### PR TITLE
better FreeBSD support

### PR DESCRIPTION
**Pull request by Irgendwer01**
Due to Microsoft not shipping LWJGL libraries and natives for FreeBSD, the user needs to install LWJGL himself over the ports on FreeBSD, this PR makes if FreeBSD is found to use the native system libraries to make it possible playing MC under FreeBSD, it's not the cleanest fix but enough for now.

### CODE REVIEW

**Comment by glowiak** [comment]
Yeah it's a good fix, but it will only work for <1.12.2. Yet the lwjgl3 port didn't yet get merged into Ports, but we can ship the libraries ourselves meantime

**Review by glowiak — APPROVED** [review]
you should add this, if it's 1.12.2 or lower

**Comment by glowiak** [comment]
i've put some libraries to git, yet 64bit only (there is no 32bit freebsd build): https://codeberg.org/glowiak/lwjgl-libs

File: launcher/minecraft/MinecraftInstance.cpp [hunk 1]
```diff
@@ -213,8 +213,12 @@ QString MinecraftInstance::binRoot() const
 
 QString MinecraftInstance::getNativePath() const
 {
+#if defined(Os_FreeBSD)
```
Comment (Scrumplex) [inline launcher/minecraft/MinecraftInstance.cpp]:
```suggestion
#ifdef Q_OS_FREEBSD
```

**Review by Scrumplex — APPROVED** [review]
Looks good, apart from this

File: launcher/minecraft/MinecraftInstance.cpp (hunk 1 above)
Comment (Irgendwer01) [reply launcher/minecraft/MinecraftInstance.cpp]: @Scrumplex done

**Comment by Irgendwer01** [comment]
@glowiak wouldn't it be better to ship the libraries with the LWJGL jar like it's done for all other Operating Systems?
If you could can ship it with the LWJGL jar, then I could can rebase my PR to make it download the FreeBSD jars instead of using those from the ports tree

**Review by txtsd — APPROVED** [review]

**Review by ZekeZDev — APPROVED** [review]

**Comment by glowiak** [comment]
I can. I do

**Comment by glowiak** [comment]
https://codeberg.org/glowiak/lwjgl-libs/releases/tag/jars

**Comment by glowiak** [comment]
@Irgendwer01 i did

**Pull request merged**